src/plugins/intel_gna/legacy/src/convert_function_to_cnn_network.cpp

```diff
@@ -1968,15 +1970,29 @@ CNNLayerCreator::CNNLayerCreator(const std::shared_ptr<::ngraph::Node>& node) :
                        });
 }
 
-CNNLayerPtr CNNLayerCreator::create() {
+CNNLayerPtr CNNLayerCreator::create(const std::shared_ptr<::ngraph::Node>& origin) {
+    node = origin;  // node used by node->visit_attributes(..) > AttributeVisitor::on_attribute(..)
```

Comment:
Why cannot we remove node field of the class and operate only on origin?

Reply:
used by CNNLayerCreator::on_adapter(..), which in turn is used by visit_attributes(..) > on_attribute(...) generic mechanism.